File: CPP/Program/Cloth.cpp
#include <iostream>
#include <string>

using namespace std;
class Product;

class Cloth : public Product {
private:
    string size;
    string material;
    string gender;

public:
    // Default constructor
    Cloth() : size(""), material(""), gender("") {}

    // Parameterized constructor
    Cloth(const string size, const string material, const string gender)
        : size(size), material(material), gender(gender) {}

    // Getter and Setter methods for size
    string get_Size() const {
        return size;
    }

    void set_Size(const string size) {
        this->size = size;
    }

    // Getter and Setter methods for material
    string get_Material() const {
        return material;
    }

    void set_Material(const string material) {
        this->material = material;
    }

    // Getter and Setter methods for gender
    string get_Gender() const {
        return gender;
    }

    void set_Gender(const string gender) {
        this->gender = gender;
    }
    ~Cloth(){};
};

File: PHP/Program/Cloth.php
<?php
require_once('Product.php');
class Cloth extends Product
{
    private $size;
    private $material;
    private $gender;

    public function __construct($size = "", $material="", $gender="")
    {
        parent::__construct("", "", "", "");
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    public function get_Size()
    {
        return $this->size;
    }

    public function set_Size($size)
    {
        $this->size = $size;
    }

    public function get_Material()
    {
        return $this->material;
    }

    public function set_Material($material)
    {
        $this->material = $material;
    }

    public function get_Gender()
    {
        return $this->gender;
    }

    public function set_Gender($gender)
    {
        $this->gender = $gender;
    }
}
?>